spec/mongo/client_spec.rb

```diff
@@ -444,6 +444,25 @@
         end
       end
 
+      context 'when max_pool_size and min_pool_size are both nil' do
+
+        let(:client) do
+          described_class.new(['127.0.0.1:27017'], options)
+        end
+
+        let(:options) do
+          {
+              :min_pool_size => nil,
+              :max_pool_size => nil
+          }
+        end
+
+        it 'does not set either option' do
+          expect(client.options[:max_pool_size]).to be_nil
+          expect(client.options[:min_pool_size]).to be_nil
+        end
```

Comment:
I think I'll leave it as verifying that neither option value is set, just to be sure that we don't end up with options erroneously being set. I'd like to be a little more specific than just making sure an exception is not raised.